roboflow/core/workspace.py

```diff
@@ -99,6 +100,116 @@ def clip_compare(
             comparisons = sorted(comparisons, key=lambda item: -list(item.values())[0])
         return comparisons
 
+    def two_stage(
+        self,
+        image: str = "",
+        first_stage_model_name: str = "",
+        first_stage_model_version: int = 0,
+        second_stage_model_name: str = "",
+        second_stage_model_version: int = 0,
+    ) -> dict:
+        """for each prediction in the first stage detection, perform detection with the second stage model
+        @params:
+            image: (str) = name of the image to be processed
+            first_stage_model: (str) = URL path to the first stage detection model
+            first_stage_model_version: (int) = version number for the first stage model
+            second_stage_mode: (str) = URL path to the second stage detection model
+            second_stage_model_version: (int) = version number for the second stage model
+            returns: (dict) = a json obj containing
+        """
+        results = []
+
+        # create PIL image for cropping
+        pil_image = Image.open(image).convert("RGB")
+
+        # grab first and second stage model from project
+        stage_one_project = self.project(first_stage_model_name)
+        stage_one_model = stage_one_project.version(first_stage_model_version).model
+        stage_two_project = self.project(second_stage_model_name)
+        stage_two_model = stage_two_project.version(second_stage_model_version).model
+
+        print(self.project(first_stage_model_name))
+
+        # perform first inference
+        predictions = stage_one_model.predict(image)
+
+        if stage_one_project.type == "object-detection":
+            # interact with each detected object from stage one inference results
+            for boundingbox in predictions:
+                # rip bounding box coordinates from json1
+                # note: infer returns center points of box as (x,y) and width, height
+                # ----- but pillow crop requires the top left and bottom right points to crop
+                box = (
+                    boundingbox["x"] - boundingbox["width"] / 2,
+                    boundingbox["y"] - boundingbox["height"] / 2,
+                    boundingbox["x"] + boundingbox["width"] / 2,
+                    boundingbox["y"] + boundingbox["height"] / 2,
+                )
+
+                # create a new cropped image using the first stage prediction coordinates (for each box!)
+                croppedImg = pil_image.crop(box)
+                croppedImg.save("./temp.png")
+
+                # capture results of second stage inference from cropped image
+                results.append(stage_two_model.predict("./temp.png")[0])
+        elif stage_one_project.type == "classification":
+            print(
+                "please use an object detection model--can only perform two stage with bounding box results"
+            )
+
+        return results
```

Comment:
Anything else we want to do for post proc of the results? I.E. scale them back up to the dimensions of the first image?

If this is for classification only - add a model2 restrictor?